core/services/offchainreporting/block_translator.go

```diff
@@ -0,0 +1,79 @@
+package offchainreporting
+
+import (
+	"math/big"
+
+	"github.com/smartcontractkit/chainlink/core/chains"
+)
+
+// BlockTranslator converts emitted block numbers (from block.number) into a
+// block number range suitable for query in FilterLogs
+type BlockTranslator interface {
+	NumberToQueryRange(changedInBlock uint64) (fromBlock *big.Int, toBlock *big.Int)
+	Start()
+	Close()
```

Comment:
all Start and Close implementations for this are empty at the moment - could remove?

Reply:
Can do, I deliberately left these to make room for a followup implementation which will come with performance improvements.

But you are right, this can be easily added later. Will remove.